sklearn/ensemble/bagging.py

```diff
@@ -454,14 +483,14 @@ def _validate_estimator(self):
         super(BaggingClassifier, self)._validate_estimator(
             default=DecisionTreeClassifier())
 
-    def _set_oob_score(self, X, y):
+    def _set_oob_score(self, X, y, n_more_estimators):
         n_classes_ = self.n_classes_
         classes_ = self.classes_
         n_samples = y.shape[0]
 
         predictions = np.zeros((n_samples, n_classes_))
 
-        for estimator, samples, features in zip(self.estimators_,
+        for estimator, samples, features in zip(self.estimators_[-n_more_estimators:],
                                                 self.estimators_samples_,
                                                 self.estimators_features_):
```

Comment:
I would be in favor of raising an error for now when you want the oob scores with `warm_start=True` and this is not the first batch.

> Meh, I just checked forest.py and it seems _set_oob_score in forest is wrong upon warm restart. It reuses X, y from the last batch, making the indices invalid.

Arf :-/